### PR TITLE
Add support for Sponsored/Sponsored By CompanyName

### PR DESCRIPTION
**Pull request by B77Mills**
Add support to say that if there is a label on the content for Sponsored to change the card text to Sponsored or if has a company assigned to it as well it will say Sponsored By CompanyName

Screen Shot 2019-12-08 at 11 10 34 AM

### CODE REVIEW

File: packages/common/utils/get-sponsored-by-text.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = (node, fallbackValue) => {
+  const { labels = [] } = node;
+  const sponsored = labels.includes('Sponsored');
+  if (sponsored) {
+    return node.company ? `Sponsored by ${node.company.name}` : 'Sponsored';
```
Comment (zarathustra323) [inline packages/common/utils/get-sponsored-by-text.js]:
```js
const companyName = get(node, 'company.name');
if (sponsored && companyName) return `Sponsored by ${companyName}` : 'Sponsored';
```

File: packages/common/graphql/fragments/content-list.js [hunk 2]
```diff
@@ -13,6 +13,10 @@ fragment NewsletterContentListFragment on Content {
     src
     alt
   }
+  labels
+  company {
+    name
```
Comment (zarathustra323) [inline packages/common/graphql/fragments/content-list.js]: Return the company `id` as well

File: packages/common/utils/get-sponsored-by-text.js (hunk 1 above)
Comment (B77Mills) [reply packages/common/utils/get-sponsored-by-text.js]: @zarathustra323  This should be update . I cant do the (sponsored && companyName) as they can have a piece of content that is sponsored without a company.  I dont know why, but yeah

File: packages/common/utils/get-sponsored-by-text.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const { get } = require('@base-cms/object-path');
+
+module.exports = (node, fallbackValue = '') => {
+  const { labels = [] } = node;
+  const sponsored = labels.includes('Sponsored');
+  const companyName = get(node, 'company.name');
```
Comment (zarathustra323) [inline packages/common/utils/get-sponsored-by-text.js]: Flip L6 and L7 around

File: packages/common/utils/get-sponsored-by-text.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  if (!sponsored) return fallbackValue;
+  return companyName ? `Sponsored by ${node.company.name}` : 'Sponsored';
```
Comment (zarathustra323) [inline packages/common/utils/get-sponsored-by-text.js]: return companyName ? `Sponsored by ${companyName}` : 'Sponsored';

**Pull request merged**